components/turret.py

```diff
@@ -91,6 +91,17 @@ def _slew_to_count(self, count: int) -> None:
             self.incrementing = False
         self.current_state = self.SLEWING
 
+    def scan(self, heading):
```

Comment:
Perhaps "scan_for_target" would be a more precise name.

Reply:
It might not be scanning for a target - it could be scanning for an index.
I didn't use the word "target" because the turret has no concept of what a target is.